src/subclass/types.rs

```diff
@@ -271,6 +272,23 @@ pub trait ObjectSubclass: ObjectImpl + Sized + 'static {
         }
     }
 
+    /// Returns the implementation from an instance.
+    ///
+    /// Panics if called on an object of the wrong type.
+    fn get_impl_from_instance(obj: &Self::ParentType) -> &Self {
+        unsafe {
+            let data = Self::type_data();
+            let type_ = data.as_ref().get_type();
+            assert_ne!(type_, Type::Invalid);
+
+            assert!(obj.get_type().is_a(&type_));
```

Comment:
Wouldn't it be better/simpler to use one `assert` instead of two? Like:

```rust
assert!(type_ != Type::Invalid && obj.get_type().is_a(&type_));
```

Reply:
It's two different things we're asserting here: 1) that the type is already properly registered, 2) that the object passed in is of the correct type.

Making it a single assertion would make it a guessing game which of the two cases it is if the assertion ever fails.

Reply:
That's a good point. Nevermind then!